src/compas/scene/scene.py

```diff
@@ -108,6 +106,9 @@ def add(self, item, parent=None, **kwargs):
             The scene object associated with the item.
         """
 
+        if isinstance(item, Group):
+            raise Exception("Use Scene.add_group to add a group to the scene.")
+
```

Comment:
Letting users to interact with a `Group` instance before properly attached to `Scene` is fragile, this is added to prevent that and ask user to use `scene.add_group` instead, so the links are property established.